### ISSUE TITLE
softmaxes in optimization part 

### ISSUE THREAD

**Issue by soheejun** [antoniobarbalau/black-box-ripper#1]
Hi, thanks for the good paper and opened source. 

By the way, I have a question about the optimization part (torch_optimizer.py). 
I think the role of the optimization part is selecting the latent space that has high confidence by the teacher model. 
And, the condition of high confidence is larger than 0.9. 
If I'm right,  I think in the optimize function, the output of the classifier needs softmax operation. 
I just wonder why there is no softmax operation after getting the classifier's output (variable name softmaxes). 

If there is any wrong with my question, please tell me.
Thanks! 

Best regards.

**Comment by antoniobarbalau** [comment]
Greetings soheejun, and sorry for responding this late,

Thank you for taking interest in our work and checking out our code. We really appreciate the fact that you took your time to investigate the method. Your observation is very good, in the original version of our framework we did indeed have a softmax activation function and we optimized until 90% confidence. However, during trials, we've found that we can also perform optimization directly on the logits and use 0.9 as a threshold. In our experiments, this ensured that the black-box model was confident that the object (or information about the object, such as textures) was present in the synthesized image, without constraining the label and without minimizing features from other objects, resulting in slightly more complex samples while staying relevant to the problem. Both approaches work and you can use whichever you see fit, however, optimizing directly on the logits may require a bit of tweaking depending on the situation.

Thank you for your interest and for the question, I wish you good luck with all your work,
Antonio.
